File: Agents/Interaction/Templates/ShowMessageToUser.cs
namespace Technologai.Templates
{
    public class ShowMessageToUser : Template
    {
        internal event Action<string>? Message;

        public ShowMessageToUser(Action<string> messageCallback)
        {
            Id = "show_message_to_user";
            Description = "Show a message to the user.";
            Message += messageCallback;
        }

        public override Task<bool> Assess(Information information) => Task.FromResult(true);

        public override Task<Data?> Process(Information information)
        {
            Message?.Invoke(information?.Input?.Raw ?? string.Empty);
            return Task.FromResult((Data?)null);
        }
    }
}

File: Agents/Interaction/python_templates/show_message_to_user.py
from technologai_agent.information import Information
from technologai_agent.template import Template

class ShowMessageToUser(Template):
    def __init__(self, message_callback, **kwargs):
        super().__init__(
            id="show_message_to_user",
            description="Show a message to the user.",
            **kwargs
        )
        self.message_callback = message_callback

    async def assess(self, information: Information):
        return True

    async def process(self, information: Information):
        self.message_callback(information.input if information.input else '')
        return None